### DISCUSSION TITLE
Whiteboard feature - edges

### DISCUSSION THREAD

**Discussion by nikkujensen** [sagemathinc/cocalc#6047] [Q&A] [answered]
I have been exploring some of the Whiteboard features and there is one I don't understand. It is the `edges`. There isn't any documentation on it in the [docs](https://doc.cocalc.com/whiteboard.html) nor is there a detailed description on the [CoCalc website](https://cocalc.com/features/whiteboard).

Could someone fill me in on what this feature is? The best I can guess since it is listed under the [Jupyter Cells functionality on the website](https://cocalc.com/features/whiteboard), is that it has something to do with the execution order of cells, maybe?

**Comment by williamstein** [comment, marked as answer]
> Could someone fill me in on what this feature is?

Currently they are 100% cosmetic, i.e., just for looks.  However, as you guessed, in the future I hope to impose an execute order that is informed by edges (currently there's no "execute all" functionality).  

I would love for you to share any and all thoughts you have about the whiteboard.   I wrote it pretty quickly to have a first usable tool to experiment with before getting too bogged down in features, so there is still much left to do.   A lot of functionality involving separate panels, i.e., "Table of Contents", "Pages", and "Search", got added after the first release in order to make whiteboards more usable.

**Reply by nikkujensen** [reply]
Thanks for the reply and clarification!

I will keep thinking about it, but I don't have any concise feedback for now. I recently shared this feature of CoCalc with a friend, and he was ecstatic about its potential. He is a web developer and saw this as a fantastic idea for collaborating whiteboard-like but with executable code for a project.

**Comment by binary10** [comment]
The whiteboard is absolute anarchy.. and I do love the freedom, no matter how dangerous it is. 

<img width="1152" alt="Screen Shot 2022-09-22 at 8 30 46 PM" src="https://user-images.githubusercontent.com/3523536/191881989-242f4610-97df-4e17-ac7d-c8b6b368d719.png">

**Reply by williamstein** [reply]
Cool, and nice to see you using it in dark mode!

**Reply by williamstein** [reply]
There's a very interesting discussion regarding how to use the edges to impact and organize code evaluation here:  https://github.com/sagemathinc/cocalc/issues/6168
